### PR TITLE
check: pretty print the checker errors

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

https://asktug.com/t/topic/69143

### What is changed and how it works?

example

```
{
	"id": 0,
	"name": "table structure compatibility check",
	"desc": "check compatibility of table structure",
	"state": "fail",
	"errors": [
		{
			"severity": "warn",
			"short_error": "table `test-db`.`test-table-1` Foreign Key fk is parsed but ignored by TiDB.",
			"instruction": "please ref document: https://docs.pingcap.com/tidb/stable/mysql-compatibility#unsupported-features"
		},
		{
			"severity": "fail",
			"short_error": "table `test-db`.`test-table-1` primary/unique key does not exist",
			"instruction": "please set primary/unique key for the table"
		}
	],
	"extra": "address of db instance - :0"
}
```

```
{
		"id": 0,
		"name": "sharding table test-name consistency checking",
		"desc": "check consistency of sharding table structures",
		"state": "fail",
		"errors": [
			{
				"severity": "fail",
				"short_error": "column length mismatch (1 vs 2)",
				"left": "instance test-source table `test-db`.`test-table-1` columns [c]",
				"right": "instance test-source table `test-db`.`test-table-2` columns [c d]"
			}
		],
		"instruction": "please set same table structure for sharding tables",
		"extra": "error on sharding test-name"
	}
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: pkg/check/check.go [hunk 1]
```diff
@@ -37,15 +37,27 @@ const (
 	StateWarning State = "warn"
 )
 
+type Error struct {
+	Severity    State  `json:"severity"`
+	ShortErr    string `json:"short_error"`
+	Left        string `json:"left,omitempty"`
```
Comment (GMHDBJD) [inline pkg/check/check.go]: use a meaningful name instead of left and right in error message?

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

File: pkg/check/check.go (hunk 1 above)
Comment (lance6716) [reply pkg/check/check.go]: that's corresponding to `column length mismatch (1 vs 2)`. How about self and other?

column length mismatch (self: 1 vs other: 2)

Comment (GMHDBJD) [reply pkg/check/check.go]: make sense

File: pkg/check/check.go [hunk 2]
```diff
@@ -37,15 +37,27 @@ const (
...
 
+type Error struct {
+	Severity    State  `json:"severity"`
+	ShortErr    string `json:"short_error"`
+	Self        string `json:"self,omitempty"`
+	Other       string `json:"other,omitempty"`
+	Instruction string `json:"instruction,omitempty"`
+}
+
+func NewError(description string) *Error {
```
Comment (lichunzhu) [inline pkg/check/check.go]: What about using `NewError(description string, ...args)`? These `fmt.Sprintf` look strange 🤔

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

**Pull request merged**